sdk/cognitiveservices/azure-cognitiveservices-language-textanalytics/azure/cognitiveservices/language/textanalytics/_response_handlers.py

```diff
@@ -0,0 +1,230 @@
+
+    for idx, entity in enumerate(results):
+        if hasattr(entity, "error"):
+            results[idx] = DocumentError(id=entity.id, error=entity.error)
+        else:
+            results[idx] = DocumentEntities(
+                id=entity.id,
+                entities=[Entity._from_generated(e) for e in entity.entities],  # pylint: disable=protected-access
+                statistics=DocumentStatistics._from_generated(entity.statistics),  # pylint: disable=protected-access
+            )
+    return results
+
+
+def linked_entities_result(response, obj, response_headers):  # pylint: disable=unused-argument
+    if obj.errors:
+        combined = obj.documents + obj.errors
+        results = order_results(response, combined)
+    else:
+        results = obj.documents
+
+    for idx, entity in enumerate(results):
+        if hasattr(entity, "error"):
+            results[idx] = DocumentError(id=entity.id, error=entity.error)
+        else:
+            results[idx] = DocumentLinkedEntities(
+                id=entity.id,
+                entities=[LinkedEntity._from_generated(e) for e in entity.entities],  # pylint: disable=protected-access
+                statistics=DocumentStatistics._from_generated(entity.statistics),  # pylint: disable=protected-access
+            )
+    return results
+
+
+def key_phrases_result(response, obj, response_headers):  # pylint: disable=unused-argument
+    if obj.errors:
+        combined = obj.documents + obj.errors
+        results = order_results(response, combined)
+    else:
+        results = obj.documents
+
+    for idx, phrases in enumerate(results):
+        if hasattr(phrases, "error"):
+            results[idx] = DocumentError(id=phrases.id, error=phrases.error)
+        else:
+            results[idx] = DocumentKeyPhrases(
+                id=phrases.id,
+                key_phrases=phrases.key_phrases,
+                statistics=DocumentStatistics._from_generated(phrases.statistics),  # pylint: disable=protected-access
+            )
+    return results
+
+
+def sentiment_result(response, obj, response_headers):  # pylint: disable=unused-argument
+    if obj.errors:
+        combined = obj.documents + obj.errors
+        results = order_results(response, combined)
+    else:
+        results = obj.documents
+
+    for idx, sentiment in enumerate(results):
+        if hasattr(sentiment, "error"):
+            results[idx] = DocumentError(id=sentiment.id, error=sentiment.error)
+        else:
+            results[idx] = DocumentSentiment(
+                id=sentiment.id,
+                sentiment=sentiment.sentiment,
+                statistics=DocumentStatistics._from_generated(sentiment.statistics),  # pylint: disable=protected-access
+                document_scores=sentiment.document_scores,
+                sentences=[SentenceSentiment._from_generated(s) for s in sentiment.sentences],  # pylint: disable=protected-access
+            )
+    return results
```

Comment:
These five functions are very nearly identical, suggest factoring out the part that is the same into a decorator, something like (untested):
```
def prepare_result(func)
    def wrapper(response, obj, response_headers):
         if obj.errors:
             combined = obj.documents + obj.errors
             results = order_results(response, combined)
         else:
             results = obj.documents
  
         for idx, elt in enumerate(results):
             if hasattr(elt, "error"):
                 results[idx] = DocumentError(id=elt, error=elt)
             else:
                 results[idx] = func(elt)
         return results
    return wrapper
``` 
Then the special cases can be much simpler and not duplicate all that other code:
```
@prepare_results
def language_result(language):
    return DocumentLanguage(
        id=language.id,
        detected_language=DetectedLanguage._from_generated(language.detected_languages[0]),  # pylint: disable=protected-access
        statistics=DocumentStatistics._from_generated(language.statistics),  # pylint: disable=protected-access
    )
 ```

Reply:
Big improvement - thanks! :)